design/metadata-handling.md

```diff
@@ -597,15 +608,21 @@ metadata:
     name: nodepool-1
 spec:
   index: 0
+  templateReference: SharedRef
   metaData:
     name: machine-1-metadata
     namespace: default
   networkData:
     name: machine-1-metadata
     namespace: default
-  Metal3DataClaim:
+ claim:
     name: machine-1
     namespace: default
+  template:
+    name: nodepool-1
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+    namespace: default
+    kind: Metal3DataTemplate
```

Comment:
to be clear, those changes are not related to the proposal, they only fix the deviation between the implementation and the proposal.

Reply:
Yes. Spot on